### PR TITLE
Add support for highlighting unused imports

### PR DESCRIPTION
**Pull request by patrys**
It's a bit ugly as we have to parse strings but results in much nicer UX as it highlights the correct imports rather than the first character of the line.

### CODE REVIEW

**Comment by patrys** [comment]
Sorry, it seems that indentation got broken somehow when submitting this pull request. This is fixed now and supports more errors and warnings.

**Comment by patrys** [comment]
Just a friendly reminder that I'm hoping for a review :smile:

**Comment by steelbrain** [comment]
Sorry man @badray has been busy lately. cc @SpainTrain @Arcanemagus

**Comment by badray** [comment]
Sorry, I'm so busy lately, please ask @SteelBrain for instructions.

**Comment by SpainTrain** [comment]
@patrys Nice work.  Would it be possible to get an example file(s) that have each of the violation you have created cases for?  This would let us test it more quickly (and check for regressions in future changes)

**Comment by Arcanemagus** [comment]
@patrys as @SpainTrain mentioned, some files to test this with would make reviewing this easy :wink:.

**Comment by patrys** [comment]
I did my best to write some horrible code: https://gist.github.com/patrys/ce573b03573fedddfe14

File: lib/main.coffee [hunk 1]
```diff
@@ -71,8 +166,14 @@ module.exports =
             col = parseInt(match[2]) or 0
             toReturn.push({
               type: if match[4] is 'E' then 'Error' else 'Warning'
-              text: match[3] + '- ' + match[5]
+              text: match[3] + ' — ' + match[5]
```
Comment (Arcanemagus) [inline lib/main.coffee]: Any particular reason you switched this to an emdash? `flake8` on the command line doesn't do this, and it makes the messages take up more space.

**Comment by Arcanemagus** [comment]
This seems to work properly from what I can see, merging it in as dash/emdash isn't a critical issue :stuck_out_tongue:.

**Pull request merged**